token/program/src/processor.rs

```diff
@@ -367,6 +367,11 @@ impl Processor {
                     } else {
                         return Err(TokenError::InvalidInstruction.into());
                     }
+
+                    if account.delegate.is_some() {
```

Comment:
I'd be game to make this block unconditional.

Reply:
plus one

Reply:
57ec3c4